### PR TITLE
Use ESMF7 on yellowstone and cheyenne

### PR DESCRIPTION
**Pull request by fvitt**
modified:   config/cesm/machines/config_compilers.xml
        modified:   config/cesm/machines/config_machines.xml

WACCMX compsets need ESMF7 lib on yellowstone and cheyenne

Test suite: waccmx
Test baseline: 
Test namelist changes: 
Test status: Bit-for-bit

User interface changes?: no

Code review: Jim Edwards

### CODE REVIEW

**Comment by jedwards4b** [comment]
Please correctly fill out the PR template removing text in [] and indicating what testing you have done.

File: config/cesm/machines/config_machines.xml [hunk 1]
```diff
@@ -251,6 +251,23 @@
 	<command name="load">intel/17.0.1</command>
 	<command name="load">mkl</command>
       </modules>
+<!-- esmf_libs does not seem to be compatible with intel/17.0.1 on cheyenne
```
Comment (jedwards4b) [inline config/cesm/machines/config_machines.xml]: This should be in a bug report to cisl - was one created?

File: config/cesm/machines/config_compilers.xml [hunk 2]
```diff
@@ -995,6 +995,8 @@ using a fortran linker.
   <PFUNIT_PATH>$ENV{CESMDATAROOT}/tools/pFUnit/pFUnit3.2.8_cheyenne_Intel17.0.1_MPI_openMP</PFUNIT_PATH>
   <!-- Bug in the intel/17.0.1 compiler requires this, remove this line when compiler is updated -->
   <HAS_F2008_CONTIGUOUS>FALSE</HAS_F2008_CONTIGUOUS>
+  <ESMF_LIBDIR DEBUG="FALSE">/glade/u/home/fvitt/esmf_7_0_0_intel17/esmf/lib/libO/Linux.intel.64.mpi.default</ESMF_LIBDIR>
```
Comment (jedwards4b) [inline config/cesm/machines/config_compilers.xml]: cheyenne has an esmf module - if it is not adequate that info should be provided to cisl.

**Comment by fvitt** [comment]
This has been reported to CISL.  Does this PR need to wait until CISL fixes
the cheyenne esmf lib / intel 17 issues?

**Comment by jedwards4b** [comment]
I would rather not have anything in beta06 pointing to a personal directory - can this wait until the module issues have been addressed?

**Comment by fvitt** [comment]
This can probably wait... It depends on how fast CISL fixes the cheyenne
esmf modules.
Thanks.

**Comment by mvertens** [comment]
@fvitt - can you please ask CISL to make this a high priority issue and cc
me. We need to get WACCMX into the release and if this is slowing things
down - we need to resolve it ASAP.

**Comment by fvitt** [comment]
The esmf lib issues on cheyenne seems to be resolved.  I have updated my esmf branch.  I have tested the WACCM-X compsets on both yellowstone and cheyenne.  Can this be reviewed again and merged into the master branch if it looks OK?

Thanks.

**Comment by jedwards4b** [comment]
Francis, Please update the initial comment including removing text in [].  Thanks

**Comment by fischer-ncar** [comment]
I ran a couple of tests on yellowstone and cheyenne of the new esmf library.  No issues were found.

**Pull request merged**